DTAConfig/OptionPanels/XNAOptionsPanel.cs

```diff
@@ -54,35 +55,59 @@ protected override void ParseExtraControls(IniFile iniFile, string sectionName)
 
             foreach (var control in Children)
             {
-                if (!(control is FileSettingCheckBox controlAsFileSettingCheckBox))
-                    continue;
-
-                fileSettingCheckBoxes.Add(controlAsFileSettingCheckBox);
+                switch (control)
+                {
+                    case FileSettingCheckBox controlAsFileSettingCheckBox:
+                        customSettings.Add(controlAsFileSettingCheckBox);
+                        break;
+                    case CustomSettingFileCheckBox controlAsAdvancedFileCheckBox:
+                        customSettings.Add(controlAsAdvancedFileCheckBox);
+                        break;
+                    case CustomSettingFileDropDown controlAsFileSettingDropDown:
+                        customSettings.Add(controlAsFileSettingDropDown);
+                        break;
```

Comment:
Why not check for the `ICustomSetting` interface instead?

Reply:
As far as I remember it refused to work if I checked for `ICustomSetting`, but I'll check again just to be sure.

Reply:
Hm, seems that I remembered it wrong. Works now.